src/2026/supertrait-auto-impl.md

```diff
@@ -0,0 +1,93 @@
+
+## Motivation
+
+Trait evolution and trait hierarchy refactoring has been. As crates evolve and grow, needs to restructure trait hierarchy to accommodate richer functionality often arises, both in the standard library and the broader Rust ecosystem.
+
+Supertrait `auto impl` has been viewed as a potential solution and a promising language feature to the trait evolution problem, such that we can avoid major and elaborate rewrites in downstream crates. In essence, the feature would avoid rewrites when associate items are moved into a new supertrait, which is a common scenario that this problem concerns about.
+
+```rust
+// Before refactoring
+
+pub trait BigTrait {
+    fn method_lower_level();
+    fn method_higher_level();
+}
+
+// After refactoring
+
+pub trait BigTrait: Supertrait {
+    // This signals to the compiler that `Supertrait` implementation shall
+    // be automatically derived with the right items, so that ...
+    auto impl Supertrait;
+    fn method_higher_level();
+}
+
+pub trait Supertrait {
+    fn method_lower_level();
+}
+
+// ... this `impl BigTrait` continues to compile
+impl BigTrait for MyType {
+    fn method_lower_level() { .. }
+    // because this method is resolved to `Supertrait::method_higher_level`
+    // and used to derive the `impl Supertrait for MyType` automatically.
+    fn method_higher_level() { .. }
+}
+```
+
+### The status quo
+
+> *Elaborate in more detail about the problem you are trying to solve. This section is making the case for why this particular problem is worth prioritizing with project bandwidth. A strong status quo section will (a) identify the target audience and (b) give specifics about the problems they are facing today. Sometimes it may be useful to start sketching out how you think those problems will be addressed by your change, as well, though it's not necessary.*
+
+Supertrait `auto impl` targets library authors who needs to refactor traits or design a trait hierarchy. A recurring topic arising from this activity is that with refined trait granularity, or smaller traits in other words, also comes many required `impl`s. The problem exacerbates when it is an upstream trait receiving a refactor. While this is rightfully a breaking change, downstream crates would also have to move trait items into new `impl` blocks as the original trait is broken into smaller supertraits. This major rewrite is often undesirable for library authors as this would discourage downstream users to upgrade the library. For standard library, this often means that the changes can only land on an Edition boundary.
+
+### The next 6 months
+
+| Task                                   | Owner(s)          | Notes |
+|----------------------------------------|-------------------|-------|
+| Implementation of the language feature | @dingxiangfei2009 |       |
+| ...                                    |                   |       |
```

Comment:
I think we want to list "getting the RFC merged here", maybe "doing a lang experiment" too?

Another suggestions: "trying this out for a few standard library types" -- if you already have some candidates (`Deref`/`Receiver` IIRC?) you can list them here.